src/basic/fd-util.c

```diff
@@ -976,3 +976,17 @@ int read_nr_open(void) {
         /* If we fail, fallback to the hard-coded kernel limit of 1024 * 1024. */
         return 1024 * 1024;
 }
+
+int lockfp(int fd, int *fd_lock) {
+        if (lockf(fd, F_LOCK, 0) < 0)
```

Comment:
Nit and I'm not sure if the coding style says something about this but maybe add asserts for `fd_lock`?

Reply:
Added. Thanks.